lib/active_merchant/billing/gateways/global_collect.rb

```diff
@@ -100,6 +100,29 @@ def add_order(post, money, options, capture: false)
         post['order']['references']['invoiceData'] = {
           'invoiceNumber' => options[:invoice]
         }
+        add_airline_data(post, options) if options[:airline_data]
+      end
+
+      def add_airline_data(post, options)
+        airline_data = {}
+        airline_data['flightDate'] = options[:airline_data][:flight_date]
+        airline_data['passengerName'] = options[:airline_data][:passenger_name]
+        airline_data['code'] = options[:airline_data][:code]
+        airline_data['name'] = options[:airline_data][:name]
+
+        items = []
```

Comment:
Minor suggestion:
Consider renaming this variable to something like `flight_legs` to be more descriptive.

Reply:
What happens if only some of these `airline_data` fields are included? Do we send fields with `nil` values and can the gateway handle that?

Reply:
That's a good point. It does look like the other methods use the format
```
post['parameter'] = options[:parameter] if options[:parameter]
```

It would probably be good to follow that format here as well.

Reply:
This will work for some fields and I'll figure out which ones. The API Formerly Known As Global Collect has fairly picky requirements about what fields must or conditionally must be included and how they're validated (see https://epayments-api.developer-ingenico.com/s2sapi/v1/en_US/ruby/payments/create.html?paymentPlatform=GLOBALCOLLECT#payments-create-payload Order > additionalInput > airlineData, which will go in the docs) so I can't do this unilaterally.  I also didn't feel like we could improve on the error messages the gateway generates, so I decided to just pass theirs along; I'll write a test to demonstrate this.

Reply:
Actually it looks like the errors are the same whether we pass the missing field without a value or don't pass it at all, so maybe I can just do it unilaterally 🤔

Reply:
Okay, I have given this a whack!